tests/DotNetLightning.Core.Tests/TransactionTests.fs

```diff
@@ -326,4 +329,551 @@ let testList = testList "transaction tests" [
+                        (currentCommitmentNumber.NextCommitment())
+            addKeys (PerCommitmentSecretStore()) CommitmentNumber.FirstCommitment
+
+        let remoteRemoteCommit = {
+            Index = commitmentNumber
+            Spec = lastLocalCommitmentSpec
+            TxId = TxId <| localCommitmentTx.GetHash()
+            RemotePerCommitmentPoint = perCommitmentPoint
+        }
+        let remoteRemoteParams = {
+            DustLimitSatoshis = localParams.DustLimitSatoshis
+            MaxHTLCValueInFlightMSat = localParams.MaxHTLCValueInFlightMSat
+            ChannelReserveSatoshis = localParams.ChannelReserveSatoshis
+            HTLCMinimumMSat = localParams.HTLCMinimumMSat
+            ToSelfDelay = localParams.ToSelfDelay
+            MaxAcceptedHTLCs = localParams.MaxAcceptedHTLCs
+            Features = localParams.Features
+        }
+
+        let staticRemoteChannelConfig : StaticChannelConfig = 
+            {
+                FundingScriptCoin = fundingScriptCoin
+                AnnounceChannel = false
+                RemoteNodeId = localNodeMasterPrivKey.NodeId()
+                Network = Network.RegTest
+                IsFunder = false
+                FundingTxMinimumDepth = BlockHeightOffset32 6u
+                LocalStaticShutdownScriptPubKey = None
+                RemoteStaticShutdownScriptPubKey = None
+                LocalParams = remoteLocalParam
+                RemoteParams = remoteRemoteParams
+                RemoteChannelPubKeys = localChannelPubKeys
+            }
+
+        let remoteLocalCommit : LocalCommit =
+            {
+                Index = commitmentNumber
+                Spec = lastRemoteCommitmentSpec
+                PublishableTxs = {
+                    CommitTx = FinalizedTx remoteCommitmentTx
+                    HTLCTxs = List.Empty
+                }
+                PendingHTLCSuccessTxs = List.Empty
+            }
+
+        let remoteSavedChannelState : SavedChannelState = {
+            StaticChannelConfig = staticRemoteChannelConfig
+            RemotePerCommitmentSecrets = remoteRemotePerCommitmentSecrets
+            ShortChannelId = None
+            LocalCommit = remoteLocalCommit
+            RemoteCommit = remoteRemoteCommit 
+            LocalChanges = LocalChanges.Zero
+            RemoteChanges = RemoteChanges.Zero
+        }
+
+        let remoteSpendingLocalCommitmentRes =
+            ClosingHelpers.HandleFundingTxSpent
+                remoteSavedChannelState
+                None
+                remoteChannelPrivKeys
+                localCommitmentTx
+
+        let outputsSumFromRemoteSpendingLocalCommitment =
+            (Result.deref remoteSpendingLocalCommitmentRes.MainOutput)
+                .SendAll(remoteDestPubKey.WitHash.ScriptPubKey)
+                .SendFees(Money.Zero)
+                .BuildTransaction(true)
+                .Outputs
+                |> Seq.sumBy (fun output -> output.Value)
```

Comment:
Must be better if you pick the one you intended instead of summing up?

```fsharp
let dest = remoteDestPubKey.WItHash.ScriptPubKey
// ...
    Seq.pick (fun output -> if output.ScriptPubKey = dest then Some (output.Value) else None) // instead of Seq.filter
```

Reply:
Done.